### DISCUSSION TITLE
Run directly on devices?

### DISCUSSION THREAD

**Discussion by GeorgeHahn** [marcus-crane/october#83] [Ideas]
It strikes me that October could run directly on Kobo devices. This would allow for more frequent synchronization, give a great user experience, and bypass the issues intrinsic to interfacing with hardware from users' computers.

Projects like [Kobowriter](https://github.com/olup/kobowriter), [Kobo-UNCaGED](https://github.com/shermp/Kobo-UNCaGED), and others are written in Go and run directly on devices. The [Koblime](https://kobli.me/) project has solutions for other parts of the story: it shows that udev rules can be used to automatically sync when wifi connects, and it has a slick browser-based installer.

Looking at all of these parts, I'm confident that it's possible build a Readwise sync tool that would run directly on devices without requiring a computer connection. What do you think?

**Comment by marcus-crane** [comment]
Ah, hey.

This had come up from another user before and admittedly I'm a bit hesitant to expand the scope?

To elaborate a bit more, October started as a sort of a stop gap for myself that just had to be "good enough".

While I did open source it for others, and hacking away at it can be fun in the same way as I suppose gardening is, I'm wary of expanding it until it unnecessarily replicates functionality found in other places like Calibre etc.

I'd also mention that lately I've been reading in Readwise Reader directly rather than on my Kobo which kind of eliminates the entire class of problem instead of trying to make a better stopgap.

That's not to say I've necessarily ditched my Kobo but often I get frustrated with the highlighting experience. I think the on-device Kepub reader has a bug where some of the books will have sub-sentence spans (in the underlying document) that the viewer will refuse to create a highlight for if you cross from one text fragment to another. There are a non-zero number of things I've tried to highlight on my Kobo that it won't let me which kind of defeats the whole point of using it to highlight 😅

As far as syncing from Kobo to Readwise wirelessly, I believe KOReader does have a built in (unofficial) Readwise integration but I've never tried it myself.

I also haven't played with Koblime but it seems to be headed in an interesting direction. It would be nice if there were something open source but if it works then all the better for the ecosystem.

Anyway, as of late I've been kept pretty busy with work and general health but maybe one day if I ever get the itch to do something lower level then who knows hah.

Perhaps someone in the community can pave a more advanced way forward 🤷‍♀️
